catalog/db_batch_entry_read.go

```diff
@@ -0,0 +1,166 @@
+	startTime time.Time
+	pathList  []pathRequest
+}
+
+type bufferingKey struct {
+	repository string
+	ref        Ref
+}
+
+type batchReadMessage struct {
+	key   bufferingKey
+	batch []pathRequest
+}
+
+func (c *cataloger) initBatchEntryReader() {
+	go c.readOrchestrator()
+	c.wg.Add(1)
+	for i := 0; i < ReadersNum; i++ {
+		go c.readEntriesBatch()
+		c.wg.Add(1)
+	}
+}
+
+func (c *cataloger) dbBatchEntryRead(repository, path string, ref Ref) (*Entry, error) {
+	replyChan := make(chan readResponse, 1) // channel closed by readEntriesBatch
+	request := &readRequest{
+		bufferingKey{repository, ref},
+		pathRequest{path, replyChan},
+	}
+	c.readEntryRequestChan <- request
+	select {
+	case response := <-replyChan:
+		return response.entry, response.err
+	case <-time.After(ReadEntryTimeout):
+		return nil, ErrReadEntryTimeout
+	}
+}
+
+func (c *cataloger) readOrchestrator() {
+	defer c.wg.Done()
+	bufferingMap := make(map[bufferingKey]*readBatch)
+	timer := time.NewTimer(0)
+	for {
+		if len(bufferingMap) > 0 {
+			timer.Reset(ScanTimeout)
+		}
+		select {
+		case request, moreEntries := <-c.readEntryRequestChan:
+			if !moreEntries {
+				return // shutdown
+			}
+			batch, exists := bufferingMap[request.bufKey]
+			if !exists {
+				batch = &readBatch{
+					startTime: time.Now(),
+					pathList:  make([]pathRequest, 0, MaxEntriesInRequest),
+				}
+				bufferingMap[request.bufKey] = batch
+			}
+			batch.pathList = append(batch.pathList, request.pathReq)
+		case <-timer.C:
+		}
+		// send pending batches that are either full, or passed the timeout
+		for k, v := range bufferingMap {
+			if len(v.pathList) == MaxEntriesInRequest || time.Since(v.startTime) > waitTimeout {
+				c.entriesReadBatchChan <- batchReadMessage{k, v.pathList}
+				delete(bufferingMap, k)
+			}
+		}
+	}
+}
+
+func (c *cataloger) readEntriesBatch() {
+	defer c.wg.Done()
+	for {
+		message, more := <-c.entriesReadBatchChan
+		if !more {
+			return
+		}
+		ctx := context.Background()
+		retInterface, err := c.db.Transact(func(tx db.Tx) (interface{}, error) {
+			var entList []*Entry
+			bufKey := message.key
+			pathReqList := message.batch
+			branchID, err := c.getBranchIDCache(tx, bufKey.repository, bufKey.ref.Branch)
+			if err != nil {
+				return entList, err
+			}
+
+			lineage, err := getLineage(tx, branchID, bufKey.ref.CommitID)
+			if err != nil {
+				return entList, fmt.Errorf("get lineage: %w", err)
+			}
+
+			p := make([]string, len(pathReqList))
+			for i, s := range pathReqList {
+				p[i] = s.path
+			}
+			//pathInExper := "('" + strings.Join(p, "','") + "')"
+			inExper := sq.Select("path", "physical_address", "creation_date", "size", "checksum", "metadata", "is_expired").
+				FromSelect(sqEntriesLineage(branchID, bufKey.ref.CommitID, lineage), "entries").
+				//Where("path in " + pathInExper + " and not is_deleted")
+				Where(sq.And{sq.Eq{"path": p}, sq.Expr("not is_deleted")})
+			sql, args, err := inExper.PlaceholderFormat(sq.Dollar).ToSql()
```

Comment:
Consider just using (at the top of the file)
```go
var psql = sq.StatementBuilder.PlaceholderFormat(sq.Dollar)
```
and then
```go
inExpr := psql.Select(...)...

Reply:
I stopped using this approach because I can not use DebugSqlizer